modules/sovrnBidAdapter.js

```diff
@@ -22,48 +24,58 @@ export const spec = {
    * @return object of parameters for Prebid AJAX request
    */
   buildRequests: function(bidReqs, bidderRequest) {
-    const loc = utils.getTopWindowLocation();
-    let sovrnImps = [];
-    let iv;
-    utils._each(bidReqs, function (bid) {
-      iv = iv || utils.getBidIdParameter('iv', bid.params);
-      sovrnImps.push({
-        id: bid.bidId,
-        banner: { w: 1, h: 1 },
-        tagid: String(utils.getBidIdParameter('tagid', bid.params)),
-        bidfloor: utils.getBidIdParameter('bidfloor', bid.params)
+    try {
+      const loc = utils.getTopWindowLocation();
+      let sovrnImps = [];
+      let iv;
+      utils._each(bidReqs, function (bid) {
+        iv = iv || utils.getBidIdParameter('iv', bid.params);
+        bid.sizes = ((utils.isArray(bid.sizes) && utils.isArray(bid.sizes[0])) ? bid.sizes : [bid.sizes])
+        bid.sizes = bid.sizes.filter(size => utils.isArray(size))
+        sovrnImps.push({
+          id: bid.bidId,
+          banner: {
+            format: bid.sizes.map(size => ({w: parseInt(size[0], 10), h: parseInt(size[1], 10)})),
```

Comment:
can we do this mapping above as well? just to colocate all of the size manipulation stuff. maybe even throw it into a fuction

Reply:
Not going to make a three line function that gets called once, but I can move it up.